modules/javafx.controls/src/test/java/test/javafx/scene/control/ControlTest.java

```diff
@@ -787,6 +787,34 @@ private void enableLogging() {
         assertNull(c.getSkin());
     }
 
+    /** verifies that Control.setSkin() calls Skin.install() JDK-8290844 */
+    @Test
+    public void setSkinCallsInstall() {
+        class SkinWithInstall extends SkinStub {
+            public boolean installed;
+
+            public SkinWithInstall(Control c) {
+                super(c);
+            }
+
+            @Override
+            public void install() {
+                installed = true;
+            }
+        }
+
+        SkinWithInstall skin = new SkinWithInstall(c);
+        c.setSkin(skin);
+        assertTrue("Control.setSkin() must call Skin.install()", skin.installed);
+    }
+
+    /** Verifies that an IllegalArgumentException is thrown when setting skin for an unrelated control JDK-8290844 */
+    @Test(expected=IllegalArgumentException.class)
+    public void skinMustCorrespondToControl() {
+        SkinStub skin = new SkinStub(new ControlStub());
+        c.setSkin(skin);
+    }
```

Comment:
Is Junit 5 availabe for this project?  If so I'd recommend writing:
```suggestion
    @Test
    public void skinMustCorrespondToControl() {
        SkinStub skin = new SkinStub(new ControlStub());
        assertThrows(IllegalArgumentException.class, () -> c.setSkin(skin));
    }
```

Reply:
We don't want to mix JUnit 4 and JUnit 5 in the same file, and it seems out of scope to change this existing test class to use Junit5.

Reply:
Are we mixing?  I believe the tests have been converted to JUnit5 some time ago, [JDK-8292678](https://bugs.openjdk.org/browse/JDK-8292678) at least in Eclipse.
I think this is a valid change.

Reply:
(and it was me who added @Test(expected=IllegalArgumentException.class) test)

Reply:
I said it wrong, the asserts can be used in both 4 and 5, so `assertThrows` can always be used now that it is available (the upgrade to JUnit 5 also updated JUnit 4 to a version which supports `assertThrows`).

Reply:
In that case, this is fine. What I meant when I said we don't want to mix 4 and 5 in the same file is that we should not use some APIs from JUnit 4 (e.g., `org.junit.Assert`) and some from JUnit 5 (e.g., `org.junit.jupiter.api.Assertions`) in the same file.